services/github/src/handlers/status.js

```diff
@@ -115,69 +244,59 @@ async function statusHandler(message) {
         }
       } catch (e) {
         if (e instanceof SyntaxError) {
-          let errorMessage = `Custom annotations artifact ${annotationsArtifactName} on task ${taskId} does not contain valid JSON.`;
-          await this.createExceptionComment({
-            debug,
-            instGithub,
-            organization,
-            repository,
-            sha,
-            error: new Error(errorMessage),
-          });
-        }
-        else {
+          createExceptionComment(
+            `Custom annotations artifact ${annotationsArtifactName} on task ${taskId} does not contain valid JSON.`,
+          );
+        } else {
           await this.monitor.reportError(e);
         }
       }
     }
 
-    const output = {
-      title: `${this.context.cfg.app.statusContext} (${event_type.split('.')[0]})`,
-      summary: `${taskDefinition.metadata.description}`,
-      text: [
-        `[${CHECKRUN_TEXT}](${taskUI(this.context.cfg.taskcluster.rootUrl, taskGroupId, taskId)})`,
-        `[${CHECKLOGS_TEXT}](${taskLogUI(this.context.cfg.taskcluster.rootUrl, runId, taskId)})`,
-        liveLogText ? `\n\`\`\`bash\n${tailLog(liveLogText)}\n\`\`\`` : '', // tails the last 250 lines of the log
-        customCheckRunText || '',
-      ].join('\n'),
-      annotations: customCheckRunAnnotations,
-    };
+    const githubCheck = new GithubCheck({
+      owner: organization,
+      repo: repository,
+      name: taskDefinition.metadata.name,
+      head_sha: sha,
+      external_id: taskId,
+      details_url: taskUI(this.context.cfg.taskcluster.rootUrl, taskGroupId, taskId),
+      status: checkRunStatus,
+      conclusion,
+
+      output_title: outputTitle || `${this.context.cfg.app.statusContext} (${event_type.split('.')[0]})`,
+      output_summary: outputSummary || taskDefinition.metadata.description,
+      output_annotations: customCheckRunAnnotations,
+      output_text: `[${CHECKRUN_TEXT}](${taskUI(this.context.cfg.taskcluster.rootUrl, taskGroupId, taskId)})`,
+    });
+
+    if (liveLogText) {
+      // doesn't make sense to include link to live log if it was not fetched
+      githubCheck.output.addText(`[${CHECKLOGS_TEXT}](${taskLogUI(this.context.cfg.taskcluster.rootUrl, runId, taskId)})`);
+      githubCheck.output.addText(`\n\`\`\`bash\n${tailLog(liveLogText)}\n\`\`\``);
+    }
+
+    if (customCheckRunText) {
+      githubCheck.output.addText(customCheckRunText);
+    }
```

Comment:
Will you please flip these two blocks? I got a note from @AaronMT about the ordering here and discussed with him that it makes sense to put the user's custom check run text above the log. Their team uses the custom check run text feature to generate info about the artifact and the build, etc. but now it's pushed all the way to the bottom of this checks page like here https://github.com/mozilla-mobile/fenix/pull/26793/checks?check_run_id=8230762755.

Reply:
oh! didn't know someone was using it :) 
Sure!

Reply:
I also thought that hiding logs link when it's not there is a good thing, but then maybe we would keep it for consistency, and later improve UI to show any log (livelog <> live_backing.log issue)

Reply:
> oh! didn't know someone was using it :)

I didn't know either! Their use case makes total sense though.



> but then maybe we would keep it for consistency, and later improve UI to show any log

👍🏻